src/_components/AuthenticateComponent.jsx

```diff
@@ -1,22 +1,73 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Route, Redirect } from 'react-router-dom';
+import { Route, Redirect, withRouter } from 'react-router-dom';
+import { Button } from 'semantic-ui-react';
+import ArtModal from '../components/common/ModalComponent';
 
-const AuthenticateComponent = ({ component: Component, isAuthenticated, ...options }) => (
+export const AuthenticateComponent = ({ component: Component, isAuthenticated, ...rest }) => (
   <Route
-    {...options}
+    {...rest}
     render={props => (
       isAuthenticated
-        ? <Component {...props} />
-        : <Redirect to={{ pathname: '/', state: { from: props.location } }} />
+        ? <Authorized {...props} />
+        : <Redirect to="/" />
     )}
   />
 );
 
 AuthenticateComponent.propTypes = {
-  location: PropTypes.object,
   component: PropTypes.func,
   isAuthenticated: PropTypes.bool
 };
 
-export default AuthenticateComponent;
+AuthenticateComponent.defaultProps = {
+  isAuthenticated: false
+};
+
+export class Authorized extends React.Component {
```

Comment:
@Arthur236 change of mind 😅...Move the `Authorized` component  to its own file. Maybe you can have `AuthenticateComponent` and `Authorized`  components be in the same folder(as they are dealing with access control) but different files

Reply:
😄 okay